### ISSUE TITLE
Sound(MIC) support

### ISSUE THREAD

**Issue by teacup-on-rockingchair** [aramg/droidcam#1]
I started playing with the application for Linux and made some minor improvements. Also I have managed to start streaming of sound using some reverse engineering and part of the code you commented out. I was wondering if you can give me some hint about the encoding use to stream the audio so I can try to enable also microphone input forwarding.

**Comment by aramg** [comment]
FYI - I've pushed a working prototype here via the netlink approach:
https://github.com/aramg/alsa-minivosc-src/tree/netlink_approach

Its been slow progress with other obligations, and lack of docs on netlink - but it seems to be working now. Any help with the missing features would be awesome.. I'll continue towards adding this to the actual droidcam repo as I get time.

**Comment by thierryg** [comment]
hi, nice to hear that, good work.

did you try using either the snd_loop or snd_dummy also drivers ?

I will try your work when it will be merge to the droidcam client ; when
will u plan to do it ?

Regards

Thierry GAYET
Mobile : +33.663.849.589
Skype : terranova44 / tux35220
Member of the GNU/Linux Fundation :
https://www.linuxfoundation.org/users/thierryg35
www.nextinnovation.org
www.apprendreindonesien.com

2015-02-14 4:18 GMT+01:00 aramg <notifications@github.com>:

**Comment by tristan-k** [comment]
Any news on this? It's almost a year now.

**Comment by halitalf** [comment]
I, too, would like some news of some sort. It _has_ been over a year.

**Comment by aramg** [comment]
All,
Unfortunately the news is that there isnt much news.
I'm busy with regular work, and whatever free time is left ends up being spent on other ideas/features, so far. Plus its summer. All I can say is I'm working hard towards gaining more free time for personal projects such as DroidCam, and will hopefully reach that goal soon enough.

**Comment by 123batman** [comment]
Another year has passed. Please, please give us the good stuff! :-)
I mean all this awesome work to make Android work as a webcam on Linux and all that is missing is mic support. And you seem to have the solution worked out already and only have to add it to the Droidcam repo. Do you still want to integrate it or have you stopped working on Droidcam?

**Comment by halitalf** [comment]
It seems he is still developing DroidCam.
![Play Screenshot](https://image.prntscr.com/image/0XKkvKZKTmeKpXkVXH2SRg.png)

**Comment by Jodelle5** [comment]
Hey!, have you stopped working on droidcam? I really want to use mic on linux.

**Comment by tberghuis** [comment]
So is it possible to get audio mic working?

**Comment by cracolo** [comment]
still not working  audio mic ?

**Comment by bigbruno** [comment]
maybe one day, but we wanna if this day not so long

**Comment by Lakedaemon** [comment]
I would have loved that too (because of Covid-19, we teachers stay at home and it would have been nice to stream video&audio through Usb to our students... I'll try to look for other android app that enables that (not much I guess) or I'll buy some hardware).

**Comment by pdvrieze** [comment]
@Lakedaemon Same here. I've fixed some build issues out of interest. I'd like audio to work, but frankly a proper headset will create much better sound (if you have it available).

**Comment by Lakedaemon** [comment]
Yeah. I found an alternative solution. I used an aux cable to connect the headphone output of my phone to the line in of my linux box, 
Then I was able to stream 720p video through usb thanks to droidcam and sound through the line in and to mux both stream in obs-studio.

Many htanks for droidcam. 
This helps my students having regular math courses during the covid19 confinement :)

**Comment by ronymesquita** [comment]
As an alternative to use audio, [WO Mic](https://wolicheng.com/womic/) can be used on Windows, Linux and Android to use the phone as a microphone.
I have tried on Ubuntu and it worked!

**Comment by aramg** [comment]
Closing this in favour of #54

It only took 6 years 🤯

**Comment by thosecars82** [comment]
Hello

Droidcam works fine in a 18.04 Ubuntu installation, connecting the phone and the computer to the same WIFI network. 
The speakers of the computer along with the mic of the phone and the camera of the phone work in case I do any videoconference. I guess that is what droidcam is used for.

Nonetheless I would like to get the following working:
- the computer's audio gets streamed to the phone by WIFI
- a bluetooth headset is connected to the phone and both the headset's microphone and the speakers work.

When I tried the following it didn't work: I connected droidcam and I played youtube videos at the same time and then the sound did not reach the phone, it stayed at the computer. Moreover, in case I connected a bluetooth headset to the phone, the mic of the headset kept playing no role at all. The only mic that worked was the phone's mic.

Does anyone know how to do this?


Thanks in advance
